src/pat/autotoc/autotoc.js

```diff
@@ -78,16 +78,21 @@ export default Base.extend({
         var activeId = null;
 
         $(self.options.levels, self.$el).each(function (i) {
-            const section = this.closest(self.options.section);
             const $level = $(this);
-            let id = $level.prop("id") ? $level.prop("id") : $(section).prop("id");
+            const $section = $level.closest(self.options.section);
+            let id = $section.prop("id") || $level.prop("id");
 
-            if (!id || $("#" + id).length > 0) {
+            if (!asTabs || !id || $(`#${id}`).length > 1) {
                 id = self.options.IDPrefix + self.name + "-" + i;
             }
 
             const tabId = `${id}-tab`;
-            $(section).attr("id", id).attr("aria-labelledby", tabId);
+
+            if (!asTabs) {
+                $level.attr("id", id).attr("aria-labelledby", tabId);
```

Comment:
I set the auto generated ID to the level item, so that its uniquely labelled. In theory, we do not have to trigger the click event anymore, because the URL hash points to the level element. What you think @thet ?